test/disjoint_pool.cpp

```diff
@@ -0,0 +1,82 @@
+// Copyright (C) 2023 Intel Corporation
+// Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "memoryPool.hpp"
+#include "pool.hpp"
+#include "provider/provider_null.h"
+#include "provider/provider_trace.h"
+#include "provider.hpp"
+#include "pool_disjoint.hpp"
+
+static usm::DisjointPool::Config poolConfig() {
```

Comment:
It might be better to use the C API you added for configuration. Then, you would use regular C API `umfPoolCreate` instead of the `poolMakeUnique` below.

Reply:
Why is it better to use `umfPoolCreate` than `poolMakeUnique` here?

Reply:
I think this test should test the public API and `poolMakeUnique` will not be public (at least there is no plan for that right now). The C++ helpers (`poolMakeUnique`, `providerMakeUnique`, etc.) were created mostly to simplify writing tests where we need to have a custom (mock) pool/provider and to simplify integration with Unified Runtime adapters.

Reply:
@igchor Could you take a look at the top-last commit?

Reply:
Looks good. Can we remove the `poolConfig` function now?

Reply:
@igchor poolConfig() is still used in `umf_test::makePoolWithOOMProvider<usm::DisjointPool>()`
https://github.com/ldorau/unified-memory-framework/blob/5f80bced56adf7a8d8f41a16bd4ad1a092dcc260/test/disjoint_pool.cpp#L90

Reply:
Would it be possible to rewrite `makePoolWithOOMProvider` so that we would use the C API as well? I would like to get rid of the `DisjointPoolConfig` entirely.

I see that `makePoolWithOOMProvider` is also used here: https://github.com/ldorau/unified-memory-framework/blob/5f80bced56adf7a8d8f41a16bd4ad1a092dcc260/test/memoryPoolAPI.cpp#L143 but I think we can remove it - it was only to 'test the test'.

Reply:
Actually, I'm working on one PR where I want to remove the DisjointPoolConfig entirely so I can rewrite the `makePoolWithOOMProvider` there. You can leave it for now.

Reply:
OK, I leave it for now.